### DISCUSSION TITLE
Updated using the Installer method to 22.05 but version still says 21.11

### DISCUSSION THREAD

**Discussion by mariaa144** [nix-community/NixOS-WSL#127] [Q&A] [answered]
I updated using the instructions here: https://github.com/nix-community/NixOS-WSL/releases

The update seemed to work fine but when I check the version it still says 21.11, even after a `wsl -t NixOS` and `wsl -d NixOS`.

```
$ nix-info -m
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.16.3-microsoft-standard-WSL2, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

$ nixos-version
21.11.20220325.d89f18a (Porcupine)
```

Byobu also shows 21.11 as the verison.

Is this expected? Did I do something wrong? Is this a bug?

**Comment by shayne** [comment, marked as answer]
What does `sudo nix-channel --list` show?

**Reply by mariaa144** [reply]
```
$ sudo nix-channel --list
nixos https://nixos.org/channels/nixos-21.11
```

**Reply by shayne** [reply]
Using flakes it shouldn't matter, but FWIW I changed mine to 22.05 and updated.

From the directory with your wsl flake.nix, have you run `nix flake update` to update the inputs of the flake?

**Reply by mariaa144** [reply]
Maybe the channel is used for the version check.  🤷

**Reply by mariaa144** [reply]
Once is change the channel and updated my version showed 22.05.

```
$ sudo nixos-version
22.05.2764.0ba2543f8c8 (Quokka)
```

```
$ nix-info -m
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.16.3-microsoft-standard-WSL2, NixOS, 22.05 (Quokka), 22.05.2764.0ba2543f8c8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(root): `"nixos-22.05.2764.0ba2543f8c8"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Reply by nzbr** [reply]
Sorry, my bad. I have added the commands for switching the nixos channel to 22.05 to the instructions on the release page. 

@shayne: Unless the system config comes from a flake (using `nixos-rebuild --flake ...`), the `nixos` channel is what's used as the `pkgs` input. NixOS-WSL is itself a flake, but the system config isn't per default (it's a regular `/etc/nixos/configuration.nix`). It imports the modules through flake-compat

**Reply by shayne** [reply]
@nzbr gotcha, thanks for the clarification. I use NixOS WSL as part of a my [nixos-config](https://github.com/shayne/nixos-config) and wasn't sure how a standard install is setup.
